include/albatross/src/indexing/group_by.hpp

```diff
@@ -71,6 +71,22 @@ template <typename KeyType, typename ValueType> class GroupedBase {
 
   std::map<KeyType, ValueType> get_map() const { return map_; }
 
+  std::pair<KeyType, ValueType> first_group() const { return *map_.begin(); }
+
+  auto min_value() const {
+    const auto value_compare = [](const auto &x, const auto &y) {
```

Comment:
You don't need to change it, but I don't think the `const` is strictly necessary here. Lambda captures are `const` by default in C++, so the `const` here isn't applied to the captures. And the fact that they have a unique and anonymous type makes it difficult for me to see how you would reassign the variable holding the lambda;

Reply:
Oh I see, so just `auto value_compare`?  I *have* been able to modify variables captured by a lambda, I can actually do so even if the function is marked `const`.  This for example outputs `1`:
```
    int i = 0;
    const auto increment = [&]() {
        ++i;
    };
    increment();
    std::cout << i << std::endl;
```

Reply:
**_Somewhat moot tangent incoming_**

That's expected behavior. In that case the captured variable is a reference which is `const`, not a reference to a `const`. Put another way the type of the captured `i` variable is `int& const` not `const int&`. It's super confusing. The end result is that lambdas can't modify their internal state by default, but can modify external state if that's how you capture it. Capturing by value wouldn't compile in that case:
```
int i = 0;
auto increment = [=]() {
    ++i; // error: increment of read-only variable 'i'
};
increment();

auto mut_increment = [=]() mutable {
    ++i; // compiles
};
mut_increment();
```

Reply:
I see, thanks for the explanation!